pkg/gardenlet/controller/shoot/shoot_care_control.go

```diff
@@ -143,10 +142,38 @@ type CareControlInterface interface {
 // implements the documented semantics for caring for Shoots. You should use an instance returned from NewDefaultCareControl()
 // for any scenario other than testing.
 func NewDefaultCareControl(clientMap clientmap.ClientMap, k8sGardenCoreInformers gardencoreinformers.Interface, secrets map[string]*corev1.Secret, imageVector imagevector.ImageVector, identity *gardencorev1beta1.Gardener, gardenClusterIdentity string, config *config.GardenletConfiguration) CareControlInterface {
-	return &defaultCareControl{clientMap, k8sGardenCoreInformers, secrets, imageVector, identity, gardenClusterIdentity, config}
+	return &defaultCareControl{
+		NewOperation,
+		NewHealthCheck,
+		NewConstraintCheck,
+		NewGarbageCollector,
+		clientMap,
+		k8sGardenCoreInformers,
+		secrets,
+		imageVector,
+		identity,
+		gardenClusterIdentity,
+		config,
+	}
 }
 
+var (
+	// NewOperation is used to create a new `operation.Operation` instance.
+	NewOperation = defaultNewOperationFunc
+	// NewHealthCheck is used to create a new Health check instance.
+	NewHealthCheck = defaultNewHealthCheck
+	// NewConstraintCheck is used to create a new Constraint check instance.
+	NewConstraintCheck = defaultNewConstraintCheck
+	// NewGarbageCollector is used to create a new Constraint check instance.
+	NewGarbageCollector = defaultNewGarbageCollector
```

Comment:
Are these `default...` variables needed? Can't you simply do something like this:

```suggestion
	// NewOperation is used to create a new `operation.Operation` instance.
	NewOperation = defaultNewOperationFunc
	// NewHealthCheck is used to create a new Health check instance.
	NewHealthCheck = care.NewHealth
	// NewConstraintCheck is used to create a new Constraint check instance.
	NewConstraintCheck = care.NewConstraint
	// NewGarbageCollector is used to create a new Constraint check instance.
	NewGarbageCollector = care.NewGarbageCollection
```

For the operation, I understand, here it's more complex. Did you decide to do it for the other functions then for similarity reasons? OK then... :)

Reply:
The reason I had to introduce these default `var`s was because of the return types. For instance, `care.NewHealth` returns an instance of `care.Health` but the `NewHealthCheck` expects a `HealthCheck` interface type -> compile error. As an alternative I could have changed `care.NewHealth` to also return the interface type but this violates Golang [best practices](https://github.com/golang/go/wiki/CodeReviewComments#interfaces).

Reply:
OK, yeah, let's keep it then. Thanks!